File: app/src/main/java/com/example/moviemvp/movie_list/MovieListPresenter.java
package com.example.moviemvp.movie_list;

import com.example.moviemvp.model.Movie;

import java.util.List;

public class MovieListPresenter implements MovieListContract.Presenter, MovieListContract.Model.OnFinishedListener {

    private MovieListContract.View movieListView;

    private MovieListContract.Model movieListModel;

    MovieListPresenter(MovieListContract.View movieListView) {
        this.movieListView = movieListView;
        movieListModel = new MovieListModel();
    }

    @Override
    public void onDestroy() {
        this.movieListView = null;
    }

    @Override
    public void requestDataFromServer() {
        movieListModel.getMovieList(this, 1);
    }

    @Override
    public void onSuccess(List<Movie> movieArrayList) {
        movieListView.onSuccess(movieArrayList);
    }

    @Override
    public void onFailure(Throwable t) {
        movieListView.onFailure(t);
    }
}

File: app/src/main/java/com/example/moviemvp/movie_list/MovieListPresenter.kt
package com.example.moviemvp.movie_list

import com.example.moviemvp.model.Movie

class MovieListPresenter internal constructor(private var movieListView: MovieListContract.View?)
    : MovieListContract.Presenter, MovieListContract.Model.OnFinishedListener {

    private val movieListModel: MovieListContract.Model

    init {

        movieListModel = MovieListModel()
    }

    override fun onDestroy() {

        this.movieListView = null
    }

    override fun requestDataFromServer() {

        movieListModel.getMovieList(this, 1)
    }

    override fun onSuccess(movieArrayList: List<Movie>) {

        movieListView!!.onSuccess(movieArrayList)
    }

    override fun onFailure(t: Throwable) {

        movieListView!!.onFailure(t)
    }
}